nautobot/core/tests/test_graphql.py

```diff
@@ -115,6 +118,48 @@ def test_execute_saved_query_with_variable(self):
         resp = execute_saved_query("gql-2", user=self.user, variables={"name": "site-1"}).to_dict()
         self.assertFalse(resp["data"].get("error"))
 
+    def test_graphql_types_registry(self):
+        """Ensure models with graphql feature are registered in the graphene_django registry."""
+        graphene_django_registry = get_global_registry()
+        for app_label, models in registry["model_features"]["graphql"].items():
+            for model_name in models:
+                model = apps.get_model(app_label=app_label, model_name=model_name)
+                self.assertIsNotNone(graphene_django_registry.get_type_for_model(model))
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"])
+    def test_graphql_url_field(self):
+        """Test the url field for all graphql types."""
+        schema = graphene_settings.SCHEMA.introspect()
+        graphql_fields = schema["__schema"]["types"][0]["fields"]
+        for graphql_field in graphql_fields:
+            if graphql_field["type"]["kind"] == "LIST" or graphql_field["name"] == "content_type":
+                continue
+            with self.subTest(f"Testing graphql url field for {graphql_field['name']}"):
+                graphene_object_type_definition = graphene_settings.SCHEMA.get_type(graphql_field["type"]["name"])
+
+                # simple check for url field in type definition
+                self.assertIn(
+                    "url", graphene_object_type_definition.fields, f"Missing url field for {graphql_field['name']}"
+                )
+
+                graphene_object_type_instance = graphene_object_type_definition.graphene_type()
+                model = graphene_object_type_instance._meta.model
+
+                # if an instance of this model exists, run a test query to retrieve the url
+                if model.objects.exists():
+                    obj = model.objects.first()
+                    query = f'{{ query: {graphql_field["name"]}(id:"{obj.pk}") {{ url }} }}'
+                    request = RequestFactory(SERVER_NAME="nautobot.example.com").post("/graphql/")
+                    request.user = self.user
+                    resp = execute_query(query, request=request).to_dict()
+                    self.assertIsNotNone(
+                        resp["data"]["query"]["url"], f"No url returned in graphql for {graphql_field['name']}"
+                    )
+                    self.assertTrue(
+                        resp["data"]["query"]["url"].endswith(f"/{obj.pk}/"),
```

Comment:
Do all the objects' URLs end with pk or do some of the objects have slug-based URLs? Asking this since this is against `develop`.